src/v/datalake/coordinator/state_machine.cc

```diff
@@ -136,8 +136,10 @@ coordinator_stm::apply_local_snapshot(raft::stm_snapshot_header, iobuf&&) {
 
 ss::future<raft::stm_snapshot>
 coordinator_stm::take_local_snapshot(ssx::semaphore_units) {
-    return ss::make_exception_future<raft::stm_snapshot>(
-      std::runtime_error{"not implemented exception"});
+    // temporarily ignore snapshots, to be fixed later.
+    // throwing here results in uncaught exceptions, so a dummy snapshot
+    // at offset 0 avoids that.
+    co_return raft::stm_snapshot::create(0, model::offset{0}, iobuf{});
```

Comment:
iiuc, now when iceberg is enabled it prevents local retention, we need to remember to fix this